src/legacy/ui/public/vis/editors/default/components/default_editor_agg.test.tsx

```diff
@@ -0,0 +1,279 @@
+  });
+
+  it('should not show description when agg is invalid', () => {
+    defaultProps.agg.brandNew = false;
+    const comp = mount(<DefaultEditorAgg {...defaultProps} />);
+
+    act(() => {
+      (comp.find('DefaultEditorAggParams').props() as DefaultEditorAggParamsProps).setValidity(
+        false
+      );
+    });
+    comp.update();
+    expect(setValidity).toBeCalledWith(false);
+
+    expect(
+      comp.find('.visEditorSidebar__aggGroupAccordionButtonContent span').exists()
+    ).toBeFalsy();
+  });
+
+  it('should show description when agg is valid', () => {
+    defaultProps.agg.brandNew = false;
+    defaultProps.agg.type = {
+      makeLabel: () => 'Agg description',
+    };
+    const comp = mount(<DefaultEditorAgg {...defaultProps} />);
+
+    act(() => {
+      (comp.find('DefaultEditorAggParams').props() as DefaultEditorAggParamsProps).setValidity(
+        true
+      );
+    });
+    comp.update();
+    expect(setValidity).toBeCalledWith(true);
+
+    expect(comp.find('.visEditorSidebar__aggGroupAccordionButtonContent span').text()).toBe(
+      'Agg description'
+    );
+  });
+
+  it('should call setTouched when accordion is collapsed', () => {
+    const comp = mount(<DefaultEditorAgg {...defaultProps} />);
+    expect(defaultProps.setTouched).toBeCalledTimes(0);
+
+    expect(comp.find('.euiAccordion-isOpen').exists()).toBeTruthy();
+    comp.find('.euiAccordion__button').simulate('click');
+
+    expect(defaultProps.setTouched).toBeCalledWith(true);
+  });
+
+  it('should call setValidity inside onSetValidity', () => {
+    const comp = mount(<DefaultEditorAgg {...defaultProps} />);
+
+    act(() => {
+      (comp.find('DefaultEditorAggParams').props() as DefaultEditorAggParamsProps).setValidity(
+        false
+      );
+    });
+
+    expect(setValidity).toBeCalledWith(false);
+
+    expect(
+      comp.find('.visEditorSidebar__aggGroupAccordionButtonContent span').exists()
+    ).toBeFalsy();
+  });
+
+  it('should add schema component', () => {
+    defaultProps.agg.schema = {
+      editorComponent: () => <div className="schemaComponent"></div>,
+    };
+    const comp = mount(<DefaultEditorAgg {...defaultProps} />);
+
+    expect(comp.find('.schemaComponent').exists()).toBeTruthy();
+  });
+
+  describe('agg actions', () => {
+    beforeEach(() => {
+      defaultProps.agg.enabled = true;
+    });
+
+    it('should not have actions', () => {
+      const comp = shallow(<DefaultEditorAgg {...defaultProps} />);
+      const actions = shallow(comp.prop('extraAction'));
+
+      expect(actions.children().exists()).toBeFalsy();
+    });
+
+    it('should have disable and remove actions', () => {
+      defaultProps.isRemovable = true;
+      const comp = shallow(<DefaultEditorAgg {...defaultProps} />);
+      const actions = shallow(comp.prop('extraAction'));
+
+      expect(actions.children().length).toBe(2);
+      expect(actions.childAt(0).props()).toHaveProperty('content', 'Disable aggregation');
```

Comment:
To have more stable tests, we could also use data-test-subj here instead of the text

Reply:
Done.